### PR TITLE
Improve cleanup of old parking tags

### LINKED ISSUE #4534: Street parking overlay creates duplicated tags
The street parking overlay seems to tag ways with duplicated information for `:left` or `:right` and `:both`. From what I have seen so far, either the `:right` or `:left` key appears to duplicate the value of the `:both` key. I did not find any inconsistent information between these keys.

I do not know how to reproduce this issue, but I found many examples of this type of tagging on osmose, which reports this issue as "[Too many parking:lane:[side]](https://osmose.openstreetmap.fr/en/issues/open?item=3161&class=31614)". Listed below are some ways that now have duplicated `parking:condition` and/or `parking:lane` tags after they were edited using StreetComplete: 

- https://www.openstreetmap.org/way/1104454759/history
- https://www.openstreetmap.org/way/22943015/history
- https://www.openstreetmap.org/way/1104455061/history
- https://www.openstreetmap.org/way/1105429683/history
- https://www.openstreetmap.org/way/1104422600/history

Most of the changesets that introduced this type of tagging were created using v47.1 and v47.2 of StreetComplete.

### PR DESCRIPTION
**Pull request by tapetis**
Closes #4534.

I have not seen any problems while testing and the new test cases all work as well. However, I am not really sure if the new code covers all possible parking situations and user selections.

### CODE REVIEW

**Comment by westnordost** [comment]
Very nice, you even added all those new test cases! Will review asap

File: app/src/test/java/de/westnordost/streetcomplete/osm/street_parking/StreetParkingKtTest.kt [hunk 1]
```diff
@@ -160,7 +179,125 @@ class StreetParkingTest {
...
+            LeftAndRightStreetParking(
+                StreetParkingPositionAndOrientation(ParkingOrientation.PARALLEL, ParkingPosition.ON_STREET),
+                StreetParkingPositionAndOrientation(ParkingOrientation.PARALLEL, ParkingPosition.ON_STREET)
+            ),
+            arrayOf(
+                StringMapEntryModify("parking:lane:both", "parallel", "parallel"),
+                StringMapEntryAdd("parking:lane:both:parallel", "on_street"),
+                StringMapEntryDelete("parking:lane:left:parallel", "on_street"),
+                StringMapEntryDelete("parking:lane:right:parallel", "on_kerb"),
+                StringMapEntryDelete("parking:condition:right", "customers")
```
Comment (westnordost) [inline app/src/test/java/de/westnordost/streetcomplete/osm/street_parking/StreetParkingKtTest.kt:295]: So, here I would expect the `parking:condition:right` to be deleted because the right side changed from 
parallel on-kerb parking to parallel on-street parking. (I added that line to the test).

Or am I missing something?

File: app/src/main/java/de/westnordost/streetcomplete/osm/street_parking/StreetParking.kt [hunk 2]
```diff
@@ -81,19 +81,32 @@ private val ParkingPosition.estimatedWidthOnRoadFactor: Float get() = when (this
...
+    // parking:condition:<left/right/both>
+    val conditionRight = right?.toOsmConditionValue()
+    val conditionLeft = left?.toOsmConditionValue()
+
+    /* Clean up previous parking tags when the associated side is changed. This can include removing
+     * parking:condition:*, which is a bit peculiar because most values are not even set in this
+     * function. But on the other hand, when the physical layout of the parking changes (=redesign
+     * of the street layout and furniture), the condition may very well change too, so better delete
+     * it to be on the safe side. (It is better to have no data than to have wrong data.) */
+    val keysToRemove = buildList {
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/osm/street_parking/StreetParking.kt:93]: In general I like to avoid such Kotlin constructs that implicitly change what `this` is because it adds a layer of confusion and a source of error (you need the IDE to help you display what `this` is or know the documentation by heart)

File: app/src/main/java/de/westnordost/streetcomplete/osm/street_parking/StreetParking.kt [hunk 3]
```diff
@@ -81,19 +81,32 @@ private val ParkingPosition.estimatedWidthOnRoadFactor: Float get() = when (this
...
+        if (right == left) {
+            add(Regex("^parking:lane:.*"))
+            if (conditionLeft != null && conditionRight == conditionLeft) {
+                add(Regex("^parking:condition:.*"))
             }
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/osm/street_parking/StreetParking.kt:98]: Huh? Everything is removed when left and right are the same? And something with conditions?

I don't quite understand this, shouldn't this if-else-block be about checking whether each side **changed** only?

**Comment by westnordost** [comment]
I added more granular parsing to enable implementation of the suggestions made here https://github.com/streetcomplete/StreetComplete/issues/4501#issuecomment-1290420809

They write that StreetComplete should also not delete parking tags for the side that is changed **if** it is merely supplemented to be complete tagging rather than changing anything.

**Comment by westnordost** [comment]
I'm currently looking into the code for that and maybe changing something.

**Comment by westnordost** [comment]
> Is it even possible to delete an already existing side using the app?

No

**Comment by westnordost** [comment]
Hum, I spotted some other issue while adding some tests but I forgot what it was. It was probably some edge case to the "do clear the tags if something changed". But anyway, I guess I should revert/redo some of all this to account for https://github.com/streetcomplete/StreetComplete/pull/4549#issuecomment-1290757629 ... or at least think a bit more about this first.

**Comment by westnordost** [comment]
Hmm, I thought about it some more and I think it's better to not clear the whole `parking:*` space but only replace/remove the tags set by that quest. I'll work on this now.

**Comment by westnordost** [comment]
But this PR now contains so much stuff for the conditional removal of subtags that I better start fresh and only selectively copy over some things from this PR
